src/xrpld/app/tx/detail/OfferStream.cpp

```diff
@@ -288,6 +292,17 @@ TOfferStreamBase<TIn, TOut>::step()
             continue;
         }
 
+        if (entry->isFieldPresent(sfDomainID) &&
+            !permissionedDEX::offerInDomain(
+                view_, entry->key(), entry->getFieldH256(sfDomainID), j_))
+        {
+            JLOG(j_.trace())
+                << "Removing offer no longer in domain " << entry->key();
+            permRmOffer(entry->key());
```

Comment:
Please add test which check that offer is removed if not in domain and and remain existing if in.